### PR TITLE
Polishing MIMs for ADHS, DO, Log Collector and OsConfig Configuration

### PR DESCRIPTION
**Pull request by MariusNi**
- [X] I have read the [contribution guidelines](https://github.com/Azure/azure-osconfig/blob/main/CONTRIBUTING.md).
- [X] All unit tests are passing.
- [X] I have merged the latest `main` branch prior to this PR submission.
- [X] I submitted this PR against the `main` branch.

### CODE REVIEW

**Review by robertschaedler3 — APPROVED** [review]
👍🏻

File: src/modules/mim/adhs.json [hunk 1]
```diff
@@ -0,0 +1,58 @@
+{
+  "name": "AdhsConfigurationModel",
+  "type": "mimModel",
+  "contents": [
+    {
+      "name": "adhsConfiguration",
```
Comment (simonjaeger) [inline src/modules/mim/adhs.json:6]: The name of the MIM component should be PascalCase.

File: src/modules/mim/deliveryoptimization.json [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+                    {
+                      "name": "none",
+                      "enumValue": 0
+                    },
+                    {
+                      "name": "azureDeviceUpdate",
+                      "enumValue": 1
+                    },
+                    {
+                      "name": "deviceDiscoverySerivce",
```
Comment (simonjaeger) [inline src/modules/mim/deliveryoptimization.json:34]: This is not spelled right. It should be `deviceDiscoveryService`, not `deviceDiscoverySerivce`.

Comment (simonjaeger) [reply src/modules/mim/deliveryoptimization.json:34]: Same for reported version.

File: src/modules/mim/osconfigconfiguration.json [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+  "type": "MimModel",
+  "contents": [
+    {
+      "name": "OsConfigConfiguration",
+      "type": "mimComponent",
+      "contents": [
+        {
+          "name": "desiredConfiguration",
+          "type": "mimObject",
+          "desired": false,
```
Comment (simonjaeger) [inline src/modules/mim/osconfigconfiguration.json:12]: Name of this MIM object is `desiredConfiguration`, but flag `desired` is set to false.

File: src/modules/mim/osconfigconfiguration.json [hunk 4]
```diff
@@ -0,0 +1,105 @@
...
+          "schema": "boolean"
+        },
+        {
+          "name": "commandLoggingEnabled",
+          "type": "mimObject",
+          "desired": true,
+          "schema": "boolean"
+        },
+        {
+          "name": "iotHubProtocol",
```
Comment (simonjaeger) [inline src/modules/mim/osconfigconfiguration.json:86]: Missing `desired` property set on this.

File: src/modules/mim/osconfigconfiguration.json [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+                  ]
+                }
+              }
+            ]
+          }
+        },
+        {
+          "name": "modelVersion",
+          "type": "mimObject",
+          "desired": true,
```
Comment (simonjaeger) [inline src/modules/mim/osconfigconfiguration.json:58]: Is this right? This is not desired, right?

File: src/modules/mim/osconfigconfiguration.json (hunk 3 above)
Comment (MariusNi) [reply src/modules/mim/osconfigconfiguration.json:12]: It's not a desired, but a reported object.

File: src/modules/mim/osconfigconfiguration.json (hunk 5 above)
Comment (MariusNi) [reply src/modules/mim/osconfigconfiguration.json:58]: Right

**Review by simonjaeger — APPROVED** [review]
LGTM

**Pull request merged**